src/core/packages/chrome/browser-internal/src/project_navigation/README.md

```diff
@@ -0,0 +1,112 @@
+|  +---------------------+   |    |     |                 |
+|                            |    |     |  +-----------+  |
++----------------------------+    +---->|  | navLinks$ |  |
+                                        |  +-----------+  |
+                                        |                 |
+                                        |  +-----------+  |
+                                        |  | collapsed$|  |
+                                        |  +-----------+  |
+                                        |                 |
+                                        +--------+--------+
+                                                 |
+                                                 | NavigationKibanaProvider
+                                                 | adapts streams
+                                                 v
+                                        +------------------+
+                                        | NavigationProvider|
+                                        |                  |
+                                        |  +-----------+   |
+                                        |  | Context   |   |
+                                        |  +-----------+   |
+                                        |                  |
+                                        +--------+--------+
+                                                 |
+                                                 | useNavigation() hook
+                                                 | consumes context
+                                                 v
+                                        +------------------+
+                                        | Navigation       |
+                                        | Component        |
+                                        |                  |
+                                        | - Renders UI     |
+                                        | - Highlights     |
+                                        |   active items   |
+                                        | - Shows          |
+                                        |   breadcrumbs    |
+                                        +------------------+
+```
+
+## Core Functions
+
+This service has three primary responsibilities:
+
+### 1. Consuming Navigation Components
+
+The service does not contain any UI. Instead, it processes navigation trees registered by plugins. This allows the navigation to be extended and customized by different parts of Kibana.
+
+-   **Plugins for Stateful Deployment**: Register navigation trees via the `addSolutionNavigation` service from `@kbn/shared-navigation-plugin`.
+-   **Serverless Plugins**: Register navigation trees via the `initNavigation` method from the serverless plugin's start contract.
+
+-   **How it works**: Plugins register their components, and the `ChromeService` selects the appropriate one to render based on the current context.
```

Comment:
I don't think plugins register the components nowadays? I thought there is a single component now and there is still an api to override it which is not used. Hopefully we can clean it up.